src/start.ts

```diff
@@ -67,11 +68,16 @@ export const handler = async (_event: any = {}): Promise<any> => {
           nodeSettings.airnodeWalletMnemonic
         );
 
-        let currentBlock: number;
-        try {
-          currentBlock = await provider.getBlockNumber();
-        } catch (error) {
-          printError(error);
+        // Fetch current block number from chain via provider
+        const retryableGetBlockNumberCall = node.utils.retryOnTimeout(
```

Comment:
You can use `go()` directly as it uses `retryOnTimeout` underneath.
[`go()`](https://github.com/api3dao/airnode/blob/e0784eb7854b07ca1a159bd44e99bb5d3f8a89fc/packages/airnode-node/src/utils/promise-utils.ts#L30) by default doesn't use a timeout, but you could wrap it to include a timeout in the options object or re-implement it:
Straight use:
```
const [err, currentBlock] = go(provider.getBlockNumber, {timeoutMs: 5000});
```
Wrapped use:
```
[Root scope]
const go = <T>(fn: () => Promise<T>, options?: PromiseOptions) => node.utils.go(fn, {...options, timeoutMs: 5000});

[In the handler function]
// inlined
const [err, currentBlock] = go(provider.getBlockNumber);

// non-inlined due to args
const [err, keeperSponsorWalletTransactionCount] = await go(() =>
                provider.getTransactionCount(
                  keeperSponsorWallet.address,
                  currentBlock
                ));
```
This gives me joy for some reason.